### PR TITLE
Fix syntax error and logging in hooks documentation sample script

### PR DESCRIPTION
**Pull request by grhaonan**
> **NOTE:** Kedro datasets are moving from `kedro.extras.datasets` to a separate `kedro-datasets` package in
> [`kedro-plugins` repository](https://github.com/kedro-org/kedro-plugins). Any changes to the dataset implementations
> should be done by opening a pull request in that repository.
## Description

typo in doc sample script and it will raise a python error 
## Development notes

removing an extra parenthese in the example script and tested in my local
## Checklist

- [ ] Read the [contributing](https://github.com/kedro-org/kedro/blob/main/CONTRIBUTING.md) guidelines
- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [ ] Added a description of this change in the [`RELEASE.md`](https://github.com/kedro-org/kedro/blob/main/RELEASE.md) file
- [ ] Added tests to cover my changes

<a href="https://gitpod.io/#https://github.com/kedro-org/kedro/pull/2257"></a>

### CODE REVIEW

**Review by stichbury — APPROVED** [review]
This looks good to me, thanks @grhaonan !

File: docs/source/hooks/examples.md [hunk 1]
```diff
@@ -30,10 +30,6 @@ class MemoryProfilingHooks:
     def __init__(self):
```
Comment (deepyaman) [inline docs/source/hooks/examples.md:30]: Can't comment this high up in the code, but...

1. Can we add the remaining imports? I think we're just missing `from kedro.framework.hooks import hook_impl`; seems silly that the docs leave out that one necessary import.
2. Can you format the imports as follows, just to be consistent with Black/isorted code style (why not):

    ```python
    import logging

    from kedro.framework.hooks import hook_impl
    from memory_profiler import memory_usage
    
    
    class MemoryProfilingHooks:
    ```
    
Some other inconsistencies, not sure what want to do about it:

* Some examples below put a comment with the filename at the top of the script.
* This example seems to be hidden in a "Click to expand" block, while others in the same doc aren't.

**Review by deepyaman — COMMENTED** [review]
LGTM overall, just some minor comments (and maybe some that are beyond the scope)

**Review by AntonyMilneQB — APPROVED** [review]
Thanks very much for doing this! Agree with @deepyaman's comments - would be nice if you could tweak a couple of other things on this page while you're at it, but it's not a biggy 🙂

File: docs/source/hooks/examples.md (hunk 1 above)
Comment (grhaonan) [reply docs/source/hooks/examples.md:29]: @deepyaman no problem, see in [12fc10f](https://github.com/kedro-org/kedro/pull/2257/commits/12fc10f9985d3cd392d597ded0f45746460623d3)

1. added
2. re-ordered
3. added for all examples (1&4 examples were missing the comment)
4. removed to be consistent with the rest examples in the same doc

Cheers

**Comment by stichbury** [comment]
Looks great, thanks @grhaonan -- I've just made a small commit to fix a trailing whitespace and I'll merge it.

**Pull request merged**